### PR TITLE
feat(submission): submission workflow modal

### PR DESCRIPTION
**Pull request by flesher1**
Adds listing submission modal to listing page.

Closes #AMLOS-623

Pending AMLOS-722 completion for wording fix.
Does not account for access restrictions for users (future ticket)

### CODE REVIEW

File: src/actions/listing.js [hunk 1]
```diff
@@ -205,3 +205,46 @@ export function fetchSimilarListings(id) {
...
+                dispatch(actions.pendForDeleteSuccess(json));
+                dispatch(fetchListing(id));
+                dispatch(fetchActivity(id));
+            }, error => {
+                dispatch(actions.pendForDeleteFailure(error));
+            });
+    };
+}
+
+export function deleteListing(id, message) {
```
Comment (irtiza-rizvi) [inline src/actions/listing.js]: When deleting a listing successfully, I can see the need for clearing some of the redux caches, such as frequently visited listings, maybe other parts of the storefront, etc.

File: src/components/shared/ListingStatusModal.js [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+                )
+
+            case 'REJECTED':
+                return(
+                    <div>
+                        <div>
+                            {listing.approvalStatusActivity.author} reviewed your request and rejected it with the message:  <b> {listing.approvalStatusActivity.description} </b>
+                            <br /> Please make the necessary changes and resubmit the applicaiton.
+                        </div>
+                        <button className="pt-button pt-intent-primary" onClick={() => this.props.dispatch(push(`/edit/${listing.id}`)) }>Edit Applicaiton</button>
```
Comment (irtiza-rizvi) [inline src/components/shared/ListingStatusModal.js]: typo `Applicaiton`

**Comment by irtiza-rizvi** [comment]
Couple minor comments:

1.  When the listing is approved, the `Publish` card in the modal isn't highlighted.  Is that desired?
2.  When clicking `Approve` for the org and AML approval steps, can either 1) the modal immediately be dismissed as the API call is made, or 2) not be dismissed at all?  Right now, what I see is the status change as the modal is disappearing (being picky, I know).

Otherwise, the listing goes through the workflow and the popup looks good.

File: src/components/shared/ListingStatusModal.js [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                    position={Position.RIGHT}
+                    >
+                        <Button className="btn btn-primary pt-intent-danger" intent={Intent.PRIMARY}>{buttonTerm}
+                        </Button>
+                        <div>
+                            <div>
+                            <h5>Provide a reason</h5>
+                            <textarea type="text" dir="auto" ref="reason" />
+                            </div>
+                            <button className="btn btn-primary btn-sm pt-popover-dismiss" onClick={buttonFunction}>{acceptTerm}</button>
```
Comment (blynch11p) [inline src/components/shared/ListingStatusModal.js]: Sorry, one more request.
Swap these two buttons.  I had a discussion with UX and button layouts will be 
[Cancel] [OK]
or in this case
[Dismiss] [Return to Owner]

Comment (flesher1) [reply src/components/shared/ListingStatusModal.js]: Should be changed.

**Comment by blynch11p** [comment]
Everything looks good :+1:

**Comment by irtiza-rizvi** [comment]
:+1: looks good

**Pull request merged**